lib/msal-common/src/client/SilentFlowClient.ts

```diff
@@ -75,6 +76,15 @@ export class SilentFlowClient extends BaseClient {
                 "SilentFlowClient:acquireCachedToken - Skipping cache because forceRefresh is true."
             );
             throw ClientAuthError.createRefreshRequiredError();
+        } else if (
+            !this.config.cacheOptions.claimsBasedCachingEnabled &&
+            !StringUtils.isEmptyObj(request.claims)
```

Comment:
Is request.claims an object or a stringified object?

Reply:
Stringified object, this will skip the cache if the claims value is a valid stringified object that isn't "{}" (empty) and claims based caching isn't enabled.